Add support default values

ybart opened the pull request:
When default value is specified for a given column in a migration, when there is no pre-existing translation on a record, trying to get the attribute will return nil instead of the default value.

The migration:

```ruby
Product.create_translation_table!(values: { type: :string, array: true, null: false, default: [] })
```

Currently, we get:

```ruby
Product.new.values
=> nil
```

I would rather expect:

```ruby
Product.new.values
=> []
```

With this PR, I get:

```ruby
Product.new.values
=> "{}"
```

This is better but needs a conversion, I did not figure how to do it yet.

parndt commented:
Hello @ybart, thanks for your contribution!  It would be great if you could write some supporting tests for this new feature please; is that a possibility?

ybart commented:
This should not be a problem. I'm doing that today.

ybart commented:
Code updated, rebased, and tests added.

I've improved the code so it behaves as expected with PG and added tests (returning `[]` rather than `"{}"`). The improvement shows no difference with SQLite3 (which returns `"--- []\n"` in all cases).

Since the tests are run only against SQLite3, I only tested the SQLite3 behavior.

By the way, I noticed trying to use it that `#type_cast` vas not defined for `ActiveRecord::ConnectionAdapters::Column` in AR 4.2 so I believe that `Globalize::ActiveRecord::Adapter#type_cast` is likely to be dead code.

parndt commented:
Our continuous integration works for PG and MySQL as well :smile:

ybart commented:
I did not realize that. Thanks !
